### ISSUE TITLE
(Starfield) Adding a new Alias for a Quest will result in a "null pointer"

### ISSUE THREAD

**Issue by VivecGod** [TES5Edit/TES5Edit#1259]
### **What is the version of xEdit you are using?**
SF1Edit 4.1.5 Experimental x64 ( _last public version to date while writing this bug report_ )

### **Describe the bug**
Adding 1 new Alias in a non-empty "Aliases" list ( The last "Aliases" list, not the first "VMAD > Aliases" one ) for a quest, new or copied from vanilla, and bounding this Alias to a Script Property, will result in the Papyrus log to mention the game can't bound this Alias to the property. Example of meaningful log :
`error: Property AliasMsgTxtToGetObjRefName on script myScript attached to myQuest (0B00086C) cannot be bound because <nullptr alias> (1) on quest myQuest (0B00086C) is not the right type`

### **To reproduce**

**1.** Create a new quest with DNAM - General Flags : Start Game Enabled, Starts Enabled, Run Once.

**2.** Add to this quest 2 Aliases set with :
     ► ALST - Reference Alias ID : Respectively of 0 and 1
     ► ALID - Alias Names : Respectively of prop1alias and prop2alias
     ► FNAM - Flags (sorted) : Both being Optional, Stores Text

**3.** In a texteditor, create a new "myScript" script extending "Quest", with 2 ReferenceAlias Properties having for names, respectively, "prop1alias" and "prop2alias". Then, for instance in an OnQuestInit() event, use the function ForceRefTo to fill the 2 properties with 2 different ObjectReferences ; It could be :
```
ScriptName myScript Extends Quest
  ReferenceAlias Property prop1alias Auto Const mandatory
  ReferenceAlias Property prop2alias Auto Const mandatory
    Event OnQuestInit()
      prop1alias .ForceRefTo(Game.GetPlayer() as ObjectReference)
      prop2alias .ForceRefTo(Game.GetPlayerHomeSpaceShip() as ObjectReference)
    EndEvent
```
( Compile your script with last version of Caprica: https://github.com/Orvid/Caprica )

**4.** In SF1Edit, add this script for your quest ( in VMAD > Scripts (sorted) )
Add to this quest script in SF1Edit 2 properties being prop1alias and prop2alias ; Type = Object / Object v2 FormID = the quest / Alias = 000 prop1alias ( and pick the equivalent Alias = 001 prop2alias, for prop2alias ) . Save & quit SF1Edit.

Launch the game and have a look at the error about "not bounding" and "null pointer"  in the Papyrus log in "my documents / My Games / Starfield / Log"...

...Also, you might create in SF1Edit a message object, and the property tied to it in the quest script, which you'll be triggering in your script with the "show()" function. The message text might be something like :
`Here I should be reading the Player name : <Alias=prop1alias > and here his current starship name : <Alias=prop2alias >`
As you'll then see in game, a weird thing will happen and you'll have for result that :
`Here I should be reading the Player name: [...] and here his current starship name: MyCrazyPlayerName!`

Basically, only 1 alias, the first filled, will be kept by the game which will be read it where it should have placed the last one.
This can basically be reproduced also with 3, 5, 10 or more aliases :
Only one is kept and is read at the end while all the other will just be "not bounded".

### **Expected behavior**
The expected behavior can be obtained with a simple trick ; Find any existing vanilla quest with yet enough aliases set, copy it into your .esm as a new record and modify it so you get the exact same thing as above EXCEPT you'll never be creating a new Alias ( removing an Alias or more remains possible, though and won't trigger the issue ).

Then everything will work without errors, and you'll have a correct messagebox in game :
`Here I should be reading the Player name: MyCrazyPlayerName! and here his current starship name: ShieldBreaker`
I did not tried any other setting for aliases, only with "Stores Text" as it is the simplest way to test and it's what I need currently for my own game mod... However, I am affraid anything involving Aliases would result in the same issue making many ( advanced ) scripted features unavailable. 

On my side, as mentionned above, I still can cheat in recreating vanilla quests with working Aliases lists for my project, and maybe not much modders will try to work with aliases in the further weeks and monthes (? Well, despite of its mid success, Starfield is yet another Bethesda' monster when it comes to mods, thus it's not that sure ! :P ) 
Still, it would still be very appreciated if this feature could be fixed ( or at least disabled to prevent modders searching for hours and days what's wrong with them and their code :P ) 
In the meaning time, thank you very much, Elminster, Sharlikran, Zilav & Hlp for all this tremendous work of yours, you yet nicely shared with us!

### **Screenshots**
I think the description was detailled enough, but I can provide .Esm and scripts to make the tests if needed.

**Comment by jbostrus** [comment]
The error message you posted does not match anything in the script code you posted. So I'm unclear what you think the problem actually is.

From the rest of the posting I can only presume that you put invalid data in your manually created quest. xEdit has limited hand holding and does not prevent you from putting in invalid data which will cause errors in the engine when loading it.

**Comment by ElminsterAU** [comment]
your issues come from invalid data
you need to have ALFG and ALAM 
maybe also VTCK.

I'll mark them all as required in the future.

**Comment by VivecGod** [comment]
Hi both of you, and thank you for such fast answers.

**@jbostrus**, using a scripted quest as the one I descriebed, with using the "add" feature to create an alias in SFEdit, starting from the second alias, will always result in the Papyrus log to mention that error :
`cannot be bound because <nullptr alias>`
In trying to be less confuse, sorry, the issue is the game won't take in account any of the Aliases you'll be adding from scratch with SFEdit, starting from the second one. Basically, adding aliases to a quest, for now, with SFEdit was useless to me.

**@ElminsterAU** : I am honnestly honored you also took the time to give me an answer.
My setting ALFG, etc are exaclty set the same way as for vanilla quests such as the SQ_Parent I've been used at first as template and then as a modified recorded copy.

**@Both** : I still think that the creation of any new alias in the quest list with SFEdit is the real origin of this faulty "Null Pointer" here in Papyrus log, and not the way I am setting my data ; Look, I know I could be wrong or misunderstanding what you exactly  mean : Your programing level is beyond mine, and english is not my mother tongue. I am really not here trying to make you loosing your time, plus, I absolutely get that in such a matter, yes, of course, badly set user datas will very often be the origin of any fault... And then, I perfectly understand your reaction here, as I would propbably have answered kind of the same thing : What I am coming with can be a pain in the ass to take the time to test properly, thus someone coming in yelling "my personal stuff is not working and this is your program fault" might really not be what I'd like to read when getting up the morning... :P

However, I really must insist, while maybe just staying an idiot, because I still beleive I am not mistaking here : 
I succeeded in doing what I was needing to do, thus I think I really can tell you my data are ok... 
...Except that time, I had to do it in modifying a copied vanilla quest, with an aliases list yet created, I have then cleaned and set.
The aliases in this cleaned and modifyed list ? I set them exactly just the same as I had set previously the aliases list I created from scratch, alias by alias, with SFEdit : Absolutely the same, in each field, one by one. 
I swear I have been very meticulous about that : I knew yet this workaround would work for me, before I came here to post this issue ; I wasn't expecting a direct help about it, thus why would I have had taken the time to post all of that description if I had the lesser doubt ? 
My new .esm is exactly the same as the previous faulty one, except it works. 
From there, I just can't get how it could come from my datas or settings ?

Well, I am still busy on finishing my mod as I beleive you, guys, are also busy too.
Maybe there's something wrong with me or maybe you will finally front this issue later during a better time for you to fix it :
In the meantime, if you allow me, I'll come back here with an archive including everything needed inside to check quickly this issue : I should have done that at first... I kept all my versionning backups, but my mod includes too many other things thus it would be better if I can make the simplest possible archive to share, instead of sharing my backup as it is ( it won't change anything regarding to the issue itself though ). Thank you again for your time and advices.

**Comment by ElminsterAU** [comment]
I know multiple people that create new scripted quests from scratch without issues in SF1Edit. And what you get is exactly what you see in xEdit. If your scripted quests aren't working, when done from scratch, and are working, when copied and changed, then there is a difference somewhere, and it should be visible.

**Comment by VivecGod** [comment]
You're right. I am very sorry.
I just made it all again from scratch and indeed when adding a new Alias in SFEdit, ALFG and VTCK are both greyed, while at the opposite, vanilla datas I modified have indeed these fields instancied. With "00 00 00 00" and "NULL", but still, these are instancied.
My bad. I am an idiot. Sorry :P Please, have a good day and my apologies for your loss of time.

**Comment by ElminsterAU** [comment]
No worries. That's why I've marked them as required now, so the next xEdit release will always add them automatically.

**Comment by ElminsterAU** [comment]
should be fixed in 4.1.5b with commit above

**Comment by VivecGod** [comment]
Thank you very much for your time, patience and commitment ! :)
Well, only for the record, I just discovered another one like this, but, this time, leading to a much more serious issue, as it involves savegame corruption :(
**Origin :** Adding a new Effect to an [ActiveMagicEffect] without adding and set [EFIF - Unknown] = 0 and [MUID - Unknown] = 1
**Result:** CTD during loading of any savegame made with that faultive effect ; Tested only with Script Effect. _( In the meantime, untill reloading such a savegame, the game is just working perfectly fine like for hours : That one is very sneaky and evil ! )_
